### PR TITLE
Change OverflowError message when failing on large pages

### PR DESCRIPTION
**Pull request by dargueta**
Currently, when writing a row group that's too big, all we see is an `OverflowError` exception. This is really confusing when writing string columns, as reported in #348 and #338.

This PR saves us some agony in debugging those problems by catching the exception and rethrowing it as an `IOError` with a more helpful error message.

I also included a fix in this PR for a bug I found in `thrift_structures.py` where we're throwing a `ParquetException` but not importing it first.

### CODE REVIEW

File: fastparquet/writer.py [hunk 1]
```diff
@@ -543,8 +543,13 @@ def write_column(f, data, selement, compression=None):
                                    uncompressed_page_size=l0,
                                    compressed_page_size=l1,
                                    data_page_header=dph, crc=None)
+    try:
+        write_thrift(f, ph)
+    except OverflowError as err:
+        raise IOError('Overflow error while writing page; try using row groups '
+                      'or smaller ones if you already are. Original message: ' +
```
Comment (martindurant) [inline fastparquet/writer.py]: There are always row-groups. I would be more specific "try using a smaller value for row_group_offsets".

**Comment by martindurant** [comment]
LGTM
Do you think there should be more explicit text of how to add consolidation to existing zarr/xarray data-sets, rather than creating them with consolidation turned on? 
We may also need some text around updating consolidated data-sets, but that can maybe wait to see what kind of usage people try.

File: fastparquet/writer.py (hunk 1 above)
Comment (dargueta) [reply fastparquet/writer.py]: Good point. Done.

**Comment by dargueta** [comment]
> Do you think there should be more explicit text of how to add consolidation to existing zarr/xarray data-sets, rather than creating them with consolidation turned on?

More information's great, but we should try to keep error messages short. I think it'd be better to put that in the documentation rather than make the error message longer.

**Comment by martindurant** [comment]
^ sorry my comment above seems to have gone in the wrong issue :|

**Pull request merged**